lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -437,6 +439,8 @@ signTransaction networkId resolveRewardAcct resolveAddress resolveInput (body, w
             | (addr, _) <- fromCardanoWdrls $ Cardano.txWithdrawals bodyContent
             ]
 
+        certs = take 1 $ fromCardanoCerts $ Cardano.txCertificates bodyContent
```

Comment:
`take 1` is needed as we can have stake credential registration and joining stake pool certs together and they contribute to the same witness! So we need only one

Reply:
😬 `signTransaction` should work with any transaction. Not just the ones where the cert that needs signing is the first one.

You shouldn't need to worry about duplicate key witnesses as they are represented as a set in the [ledger](https://github.com/input-output-hk/cardano-ledger/blob/ba20b1f7546b029750b52d091a1234ea4922bf90/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWitness.hs#L167-L169). If you'd like to be pedantic and the `Eq` instance is available, you could `nub` the result of `mapMaybe mkWdrlCertWitness certs`.

Reply:
ok, you are right. Great catch! I forgot that stake credential does need witness! Also `take 1` on this level is wrong. We could have in tx delegation certs that do not belong to the wallet, and are "inserted" before the wallet's cert

Reply:
yes, we handled them in decodeTx. But here we are interested into extracting stake credentials, and then answer the question whether they are our reward accounts, and if yes then add witness